cv_next/app/actions/comments/deleteComment.ts

```diff
@@ -13,6 +14,32 @@ import { logErrorWithTrace } from "@/server/base/logger";
 export const deleteComment = async (
   commentId: string
 ): Promise<Result<void, string>> => {
+  const currentIdResult = await getCurrentId();
+  if (!currentIdResult.ok) {
+    logErrorWithTrace(currentIdResult);
+    return Err(
+      "An error has occurred while deleting the comment. Please try again later."
+    );
+  }
+  const commentResult = await getCommentById(commentId);
+  if (!commentResult.ok) {
+    logErrorWithTrace(commentResult);
+    return Err(
+      "An error has occurred while deleting the comment. Please try again later."
+    );
+  }
+  const resultAdminCheck = await userIsAdmin();
+  if (
+    currentIdResult.val != commentResult.val.user_id &&
+    !resultAdminCheck.ok
+  ) {
+    logger.error(
+      `${currentIdResult.val} tried deleting someone elses comment!`
+    );
+    return Err(
+      "An error has occurred while deleting the comment. Please try again later."
+    );
+  }
```

Comment:
Please move the validation checks to a separate function (in this file),
such as you'd have:
```
const commentResult = await getCommentById(commentId);
const validationError = validateUserAction(...);
if (validationError) return validationError;
```

Nicely encapsulating the mess of validation :)

Reply:
Did it only to this file, can do to others if needed

Reply:
Wherever you think is needed, in this file it's particularly jarring